navapbc/labs-decision-support-tool
# feat: Add minimal API v0

## Ticket

https://navalabs.atlassian.net/browse/DST-605


## Changes

* Enable Fast API on startup
* Add `query` API endpoint
* Add instructions to `docs/app/getting-started.md`

## Testing

* Start app make sure `/chat` still works
* Go to http://localhost:8000/api/healthcheck
* http://localhost:8000/api/engines?user_id=myName should return `["ca-edd-web"]` for any `user_id` for now.
* `curl -X POST 'http://0.0.0.0:8000/api/query' -H 'Content-Type: application/json' -d '{ "session_id": "12", "new_session": true, "message": "unemployment insurance benefit deadlines?"}' | jq` should return a valid response
* `curl -v -X POST 'http://0.0.0.0:8000/api/query' -H 'Content-Type: application/json' -d '{ "session_id": "12", "new_session": true}'` should return an error `HTTP/1.1 422 Unprocessable Entity`

Go to https://cloud.getliteral.ai/ and check the threads:
<img width="488" alt="image" src="https://github.com/user-attachments/assets/1b0cf983-7829-443a-a7a5-5b25501674c0">
<img width="488" alt="image" src="https://github.com/user-attachments/assets/ef8221d8-6cde-4705-b5ba-008dd877e366">

* Go to http://localhost:8000/docs for API docs
